flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleEnvironment.java

```diff
@@ -171,4 +171,11 @@ ShuffleIOOwnerContext createShuffleIOOwnerContext(
 	boolean updatePartitionInfo(
 		ExecutionAttemptID consumerID,
 		PartitionInfo partitionInfo) throws IOException, InterruptedException;
+
+	/**
+	 * Gets total memory size for network buffers in bytes.
+	 *
+	 * @return network memory size in bytes.
+	 */
+	long getMemorySize();
```

Comment:
Actually we considered this issue in [FLINK-12110](https://issues.apache.org/jira/browse/FLINK-12110). 
I guess we might also need to know the memory type besides size, because we could not make an assumption of direct memory as now based on pluggable shuffle service.

Reply:
My suggestion is not covering the network resources in `ResourceProfile` atm. As I know, the `ResourceProfile` from execution only covers managed memory from table api. If so we could also cover the managed memory to match as the first step, then focus on network resources future because it might involved in how to expose/calculate network resource on RM/JM side.

Reply:
Thanks for the input, @zhijiangW. I agree with you that we do not necessarily need matching network memory at the moment. I'll update the PR accordingly.